### PR TITLE
Perform static cast + transform instead of simple copy to avoid compiler warning

### PR DESCRIPTION
**Pull request by kunaltyagi**
This should have been mainstreamed long ago :)

### CODE REVIEW

**Comment by kunaltyagi** [comment]
@jasjuang PTAL

**Comment by jasjuang** [comment]
There's a missing `;` that's preventing it from compiling. Thanks for the PR.

File: kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp [hunk 1]
```diff
@@ -171,7 +171,11 @@ knn_search(A& index, B& query, C& k_indices, D& dists, unsigned int k, F& params
   // Wrap indices vector (no data allocation)
   ::flann::Matrix<std::size_t> indices_mat(&indices[0], 1, k);
   auto ret = index.knnSearch(query, indices_mat, dists, k, params);
-  std::copy(indices.cbegin(), indices.cend(), k_indices.begin());
+  // cast appropriately
+  std::transform(indices.cbegin(),
+                 indices.cend(),
+                 k_indices.begin(),
+                 [](std::size_t x) { return static_cast<pcl::index_t>(x); });
```
Comment (mvieth) [inline kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp:178]:
```suggestion
                 [](const auto& x) { return static_cast<pcl::index_t>(x); });
```
Just an idea, probably makes no difference

**Review by mvieth — DISMISSED** [review]

File: kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp (hunk 1 above)
Comment (kunaltyagi) [reply kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp]: Makes no difference. But I made the change anyways

**Review by mvieth — APPROVED** [review]

**Review by larshg — APPROVED** [review]

**Pull request merged**